collect/queue.go

```diff
@@ -68,66 +71,110 @@ func sendBatch(batch []json.RawMessage) {
 		recordSent(len(batch))
 		return
 	}
-	var buf bytes.Buffer
-	g := gzip.NewWriter(&buf)
-	if err := json.NewEncoder(g).Encode(batch); err != nil {
-		slog.Error(err)
-		return
-	}
-	if err := g.Close(); err != nil {
-		slog.Error(err)
-		return
-	}
-	req, err := http.NewRequest("POST", tsdbURL, &buf)
-	if err != nil {
-		slog.Error(err)
-		return
-	}
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("Content-Encoding", "gzip")
-	now := time.Now()
-	resp, err := client.Do(req)
-	d := time.Since(now).Nanoseconds() / 1e6
-	if err == nil {
-		defer resp.Body.Close()
-	}
-	Add("collect.post.total_duration", Tags, d)
-	Add("collect.post.count", Tags, 1)
-	// Some problem with connecting to the server; retry later.
-	if err != nil || resp.StatusCode != http.StatusNoContent {
+	if Telnet {
```

Comment:
This diff is not very readable as it stands. Would you instead make a telnet func that does the sending, and just `if Telnet { thatFunc(); return }`?